docs/Building SpiderMonkey.md

```diff
@@ -0,0 +1,62 @@
+# Building SpiderMonkey for Embedders #
+
+Use these instructions to build your own copy of SpiderMonkey.
+
+## Prerequisites ##
+
+You will need a **C++ compiler** that can handle the C++14 standard,
+**GNU Make**, **zlib**, and **libffi**.
+These can usually be installed with a package manager.
+You will also need **ICU** of at least version 59.1.
+ICU can also usually be installed with a package manager, but the
+version might not be new enough.
+
+> **NOTE** If you can't install a new enough version of ICU, you can
+> build the copy bundled with SpiderMonkey by omitting
+> `--with-system-icu` in the build instructions below, at the cost of a
+> longer build time.
```

Comment:
I wonder if the default guidance should be to use the bundled ICU and then give a section suggesting system-icu as an optional enhancement. I suggest this so the default setup has fewer moving targets.

Reply:
Yeah, makes sense.